statServer/dist/routes/questionsRouter.js

```diff
@@ -0,0 +1,59 @@
+"use strict";
+var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
+    var desc = Object.getOwnPropertyDescriptor(m, k);
+    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
+      desc = { enumerable: true, get: function() { return m[k]; } };
+    }
+    Object.defineProperty(o, k2, desc);
+}) : (function(o, m, k, k2) {
+    if (k2 === undefined) k2 = k;
+    o[k2] = m[k];
+}));
+var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
+    Object.defineProperty(o, "default", { enumerable: true, value: v });
+}) : function(o, v) {
+    o["default"] = v;
+});
+var __importStar = (this && this.__importStar) || function (mod) {
+    if (mod && mod.__esModule) return mod;
+    var result = {};
+    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
+    __setModuleDefault(result, mod);
+    return result;
+};
+var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
+    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
+    return new (P || (P = Promise))(function (resolve, reject) {
+        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
+        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
+        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
+        step((generator = generator.apply(thisArg, _arguments || [])).next());
+    });
+};
+var __importDefault = (this && this.__importDefault) || function (mod) {
+    return (mod && mod.__esModule) ? mod : { "default": mod };
+};
+Object.defineProperty(exports, "__esModule", { value: true });
+exports.questionsRouter = void 0;
+const express_1 = __importDefault(require("express"));
+const questions_model = __importStar(require("../models/questions_model"));
+const questionsRouter = express_1.default.Router();
+exports.questionsRouter = questionsRouter;
+questionsRouter.get("/", (req, res) => __awaiter(void 0, void 0, void 0, function* () {
+    //res.header('Access-Control-Allow-Origin', 'http://localhost:3001');
+    const since = req.query.since;
+    const to = req.query.to;
+    try {
+        questions_model.getQuetionsByDates((err, albumIds) => {
+            if (err) {
+                return res.status(500).json({ "errorMessage": err.message });
+            }
+            res.status(200).json({ "data": albumIds });
+        }, { since, to });
+    }
+    catch (error) {
+        console.log(error);
+        return res.status(500).json({ "errorMessage": error });
+    }
```

Comment:
Se ve muy recargado el manejo de errores acá, el método se puede mejorar para que no se necesite if(err) y al mismo tiempo try catch:
```
export const getQuestionsByDates = (since, to): Promise<Questions[]> => {
  return new Promise((resolve, reject) => {
    // Tu código aquí...

    const queryString = `
    // ...
    `;

    db.query(queryString, (err, result) => {
      if (err) {
        reject(err); // Rechazar la promesa si hay un error
        return;
      }

      // Procesar el resultado y resolver la promesa
      // ...

      resolve(questions); // Resolver la promesa con el resultado
    });
  });
};
```
De esta manera queda simplemente:
```
try {
  const albumIds = await questions_model.getQuetionsByDates( since, to );
  res.status(200).json({ data: albumIds });
} catch (error) {
  console.log(error);
  return res.status(500).json({ errorMessage: error.message });
}
```

Reply:
Se reestructura la petición según lo solicitado.